app/controllers/register_controller.rb

```diff
@@ -27,21 +28,30 @@ def edit
   end
 
   def confirm
-    return true if params[:data_confirmed]
-    @register = get_register(params[:register])
-
-    @current_register_record = OpenRegister.record(params[:register].downcase,
-                        params[params[:register].downcase.to_sym],
-                        :beta)
-
-    if @current_register_record != nil
-      @current_register_record = convert_register_json(@current_register_record)
+    errors = @data_validator.get_form_errors(params,@registers_client.get_register(params[:register], 'beta').get_field_definitions)
+    if errors.present?
+      errors.each { |k,v| flash[k] = v[:message] }
+      @register = get_register(params[:register])
+      @form = JSON.parse(params.to_json)
+      render :new
```

Comment:
Should we also return `false` after this line, like we do on line 50?

Reply:
Doesn't seem to make a functional difference either way. Not sure what we'd be indicating by doing this?

Reply:
@danmitchell- says the other style is wrong so I will remove that `return false`